runtime/lib/app-runtime.js

```diff
@@ -457,6 +457,53 @@ AppRuntime.prototype.onVoiceCommand = function (asr, nlp, action, options) {
     })
 }
 
+/**
+ * handle mqtt 'event' message
+ * @param {string} message message string received from mqtt
+ */
+AppRuntime.prototype.onAppEvent = function (message) {
+  var data = {}
+  try {
+    data = JSON.parse(message)
+  } catch (error) {
+    data = {}
+    logger.debug('parse mqtt event message error: message -> ', message)
+    return
+  }
+  var skillId = data.appId
+  if (typeof skillId !== 'string') {
+    logger.error('Expecting data.appId exists in mqtt event message.')
+    return
+  }
+  var form = _.get(data, 'form')
+  if (typeof form !== 'string') {
+    form = _.get(this.loader.skillAttrsMap[skillId], 'defaultForm')
+  }
+  if (!form) {
+    form = 'cut'
+  }
+
+  var mockNlp = {
+    cloud: false,
+    intent: 'RokidAppCommand',
+    command: data.intent,
+    data: data.slots,
```

Comment:
might `slots: data.slots` be better?

Reply:
I don't think so. The slots of original naming is strange.

Reply:
`slots` follows intent fields of NLP. If both two works the same, the naming of `slots` shall not be changed IMO.